samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -371,47 +389,54 @@ private boolean hasActiveContainerPlacementAction(String processorId) {
     return Optional.ofNullable(this.actions.get(processorId));
   }
 
+  private void updateContainerPlacementActionStatus(ContainerPlacementMetadata metadata,
+      ContainerPlacementMessage.StatusCode statusCode, String responseMessage) {
+    metadata.setActionStatus(statusCode, responseMessage);
+    writeContainerPlacementResponseMessage(metadata.getRequestMessage(), statusCode, responseMessage);
+    LOG.info("Status updated for ContainerPlacement action: {}", metadata);
+  }
+
+  private void writeContainerPlacementResponseMessage(ContainerPlacementRequestMessage requestMessage,
+      ContainerPlacementMessage.StatusCode statusCode, String responseMessage) {
+    containerPlacementMetadataStore.writeContainerPlacementResponseMessage(
+        ContainerPlacementResponseMessage.fromContainerPlacementRequestMessage(requestMessage, statusCode,
+            responseMessage, System.currentTimeMillis()));
+  }
+
   /**
    * A valid container placement action is only issued for a running processor with a valid processor id which has no
-   * in flight container requests. Duplicate actions are handled by deduping on uuid
-   *
-   * TODO: SAMZA-2402: Disallow pending Container Placement actions in metastore on job restarts
+   * in flight container requests. Duplicate actions are handled by deduping on uuid. If there is an existing inflight
+   * request or container is pending a start, the container placement request is queued to be executed in future.
    */
```

Comment:
What does this method do?
What does it return and in which conditions does it return which output?